src/Vipps.net/Services/LoginService.cs

```diff
@@ -1,21 +1,50 @@
 ﻿using System;
 using System.Net.Http;
-using System.Security.Cryptography;
 using System.Threading;
 using System.Threading.Tasks;
-using Polly.Retry;
 using Vipps.net.Infrastructure;
 using Vipps.net.Models.Login;
 
 namespace Vipps.net.Services
 {
+    public interface IVippsLoginService
+    {
+        string GetStartLoginUri(StartLoginURIRequest startLoginUriRequest, CancellationToken cancellationToken = default);
+
+        Task<OauthTokenResponse> GetWebLoginToken(TokenRequest getTokenRequest, AuthenticationMethod authenticationMethod,
+            CancellationToken cancellationToken = default);
+
+        Task<InitCibaResponse> InitCiba(InitCibaRequest initCibaRequest,
+            AuthenticationMethod authenticationMethod, CancellationToken cancellationToken = default);
 
-    public class LoginService
+        Task<OauthTokenResponse> GetCibaTokenNoRedirect(string authReqId, 
+            AuthenticationMethod authenticationMethod, CancellationToken cancellationToken = default);
+
+        Task<OauthTokenResponse> GetCibaTokenRedirect(string code,
+            AuthenticationMethod authenticationMethod, CancellationToken cancellationToken = default);
+    }
+
+    public class LoginService : IVippsLoginService
     {
-        public static string GetStartLoginUri(StartLoginURIRequest startLoginUriRequest, CancellationToken cancellationToken = default)
+        private readonly IVippsConfigurationProvider _vippsConfigurationProvider;
+        private readonly LoginServiceClientBasic _loginServiceClientBasic;
+        private readonly LoginServiceClientPost _loginServiceClientPost; 
+        public LoginService(IVippsConfigurationProvider vippsConfigurationProvider, LoginServiceClientBasic loginServiceClientBasic, LoginServiceClientPost loginServiceClientPost)
+        {
+            _vippsConfigurationProvider = vippsConfigurationProvider;
+            _loginServiceClientBasic = loginServiceClientBasic;
+            _loginServiceClientPost = loginServiceClientPost;
+        }
+        
+        public string GetStartLoginUri(StartLoginURIRequest startLoginUriRequest, CancellationToken cancellationToken = default)
         {
-            string startLoginUri = $"{VippsConfiguration.BaseUrl}/access-management-1.0/access/oauth2/auth" +
-                                   $"?client_id={VippsConfiguration.ClientId}" +
+            var options = _vippsConfigurationProvider.GetConfiguration(); 
+            var baseUrl = options.UseTestMode 
+                ? "https://apitest.vipps.no" 
+                : "https://api.vipps.no";
```

Comment:
Same logic in `VippsHttpClient`, can refactor it out somewhere

Reply:
Refactored to a new class in the Helpers folder